service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/PlacementSubmittedEventControllerTest.java

```diff
@@ -46,93 +57,194 @@
+                eq(NOTICE_OF_PLACEMENT_ORDER_UPLOADED_TEMPLATE),
+                eq("local-authority@local-authority.com"),
+                eq(expectedParameters()),
+                eq("1"));
+
+            verify(notificationClient, never()).sendEmail(
+                eq(NOTICE_OF_PLACEMENT_ORDER_UPLOADED_TEMPLATE),
+                eq("representative@example.com"),
+                eq(expectedParameters()),
+                eq("1"));
+        }
+
+        private Map<String, Object> expectedParameters() {
+            return Map.of(
+                "respondentLastName", "Nelson",
+                "caseUrl", String.format("%s/case/%s/%s/%s", "http://fake-url", JURISDICTION, CASE_TYPE, 1L));
+        }
+
+        private Respondent respondent() {
+            return Respondent.builder()
+                .party(RespondentParty.builder()
+                    .firstName("James")
+                    .lastName("Nelson")
+                    .build())
+                .build();
+        }
+
+        private CallbackRequest callbackRequestWithEmptyCaseDetailsBefore() {
+            UUID representativeId = randomUUID();
+            Respondent respondent = respondent();
+            respondent.addRepresentative(representativeId);
+
+            return CallbackRequest.builder()
+                .caseDetails(populatedCaseDetails(representativeId, respondent))
+                .caseDetailsBefore(CaseDetails.builder().data(new HashMap<>()).build())
+                .build();
+        }
+
+        private CallbackRequest callbackRequestWithMatchingCaseDetailsBefore() {
+            UUID representativeId = randomUUID();
+            Respondent respondent = respondent();
+            respondent.addRepresentative(representativeId);
+
+            return CallbackRequest.builder()
+                .caseDetails(populatedCaseDetails(representativeId, respondent))
+                .caseDetailsBefore(populatedCaseDetails(representativeId, respondent))
+                .build();
+        }
+
+        private CaseDetails populatedCaseDetails(UUID representativeId, Respondent respondent) {
+            return CaseDetails.builder()
+                .id(1L)
+                .data(Map.of(
+                    "caseLocalAuthority", "example",
+                    "confidentialPlacements", List.of(element(Placement.builder()
+                        .childId(childElement.getId())
+                        .application(childPlacement.getValue().getApplication())
+                        .orderAndNotices(wrapElements(PlacementOrderAndNotices.builder()
+                            .type(NOTICE_OF_PLACEMENT_ORDER)
+                            .document(DocumentReference.buildFromDocument(document()))
+                            .build()))
+                        .build())),
+                    "respondents1", wrapElements(respondent),
+                    "representatives", List.of(element(representativeId, Representative.builder()
```

Comment:
`List.of` can be replaced with wrapElements

Reply:
fail to see the real benefit of this, but sure

Reply:
no I cannot do this here. I lose the id